sqlanalytics/resource_visualization.go

```diff
@@ -187,6 +190,12 @@ func ResourceVisualization() *schema.Resource {
 				return err
 			}
 
+			// Handle if visualization was not found.
+			if av == nil {
```

Comment:
this is handled generically already - every APIError with status code 404 is removing the resource... you probably need to return error the right way from Read method.

https://github.com/databrickslabs/terraform-provider-databricks/blob/master/common/resource.go#L48-L54

Reply:
Fixed!